### ISSUE TITLE
Microsoft Visual C++ Runtime Library error

### ISSUE THREAD

**Issue by FileEX** [multitheftauto/mtasa-blue#978]
**Describe the bug**
Some times mta has been crashed and show error. It's totally random and rare.

**To reproduce**

**Expected behaviour**
Fix it

**Screenshots**
![image](https://user-images.githubusercontent.com/26044224/58767706-bbdab180-858f-11e9-9797-0a234cfce242.png)

**Version**
1.5.6

**Additional context**

**Comment by Pirulax** [comment]
Hm, interesting...
My PR had the same problem(#907) if there were a lot of elements in a google::dense_hash_map(in MTA SC referred as CFastHashMap)
If i remember correctly its related to some memory issue, namely it can't allocate a new memory buffer for it's elements.
Please attach a dump, so I cant find out where the problem is.

**Comment by FileEX** [comment]
crash dump of this error has not created

**Comment by Pirulax** [comment]
neat.
my pr isn't merged yet, so nothing related to that.
without a crashdump it'll be pretty hard to investigate where the problem lies.

**Comment by ccw808** [comment]
Does `Assertion failed` ever give a crash dump?

**Comment by FileEX** [comment]
In my case, never.

**Comment by ccw808** [comment]
What about if you click "Retry" or "Ignore" ?

**Comment by Pirulax** [comment]
@ccw808 
I had the same issue, you can get it quite easily if you use CFastHashMap and fill it with elements(a few million).
The interesting this is, it doesnt really run out of memory, but for some reason if fails to allocate it.

```cpp
void clear_to_size(size_type new_num_buckets)
{
    if (!table)
    {
        table = val_info.allocate(new_num_buckets);
    }
    else
    {
        destroy_buckets(0, num_buckets);
        if (new_num_buckets != num_buckets)
        {            // resize, if necessary
            typedef base::integral_constant<bool, base::is_same<value_alloc_type, libc_allocator_with_realloc<value_type> >::value> realloc_ok;
            resize_table(num_buckets, new_num_buckets, realloc_ok());
        }
    }
    assert(table);
    fill_range_with_empty(table, table + new_num_buckets);
    num_elements = 0;
    num_deleted = 0;
    num_buckets = new_num_buckets;            // our new size
    settings.reset_thresholds(bucket_count());
}
```

**Comment by ccw808** [comment]
I would think the allocation fails because of free memory fragmentation

**Comment by Pirulax** [comment]
Yeah, thats actually a pretty good explanation, but I guess we can do nothing about it, except maybe replace it with a linked list based map, which probably would be slower due to cache misses.

**Comment by MichaelFarad** [comment]
> @ccw808 
> I had the same issue, you can get it quite easily if you use CFastHashMap and fill it with elements(a few million).
> The interesting this is, it doesnt really run out of memory, but for some reason if fails to allocate it.
> 
> ```cpp
> void clear_to_size(size_type new_num_buckets)
> {
>     if (!table)
>     {
>         table = val_info.allocate(new_num_buckets);
>     }
>     else
>     {
>         destroy_buckets(0, num_buckets);
>         if (new_num_buckets != num_buckets)
>         {            // resize, if necessary
>             typedef base::integral_constant<bool, base::is_same<value_alloc_type, libc_allocator_with_realloc<value_type> >::value> realloc_ok;
>             resize_table(num_buckets, new_num_buckets, realloc_ok());
>         }
>     }
>     assert(table);
>     fill_range_with_empty(table, table + new_num_buckets);
>     num_elements = 0;
>     num_deleted = 0;
>     num_buckets = new_num_buckets;            // our new size
>     settings.reset_thresholds(bucket_count());
> }
> ```
> 

Hi, can you please help me where to put this code? how to implement it?

**Comment by Lpsd** [comment]
@MichaelFarad what exactly are you trying to do?

**Comment by Pirulax** [comment]
ntoskrnl.exe in System32 should do it.
You will have to get the source-code for Windows 10, but that should be easy, as it can be downloaded from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).

Now, jokes aside.. You dont. (... put it anywhere, just leave it here as is.)

**Comment by MichaelFarad** [comment]
> ntoskrnl.exe in System32 should do it.
> You will have to get the source-code for Windows 10, but that should be easy, as it can be downloaded from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
> 
> Now, jokes aside.. You dont. (... put it anywhere, just leave it here as is.)

help, I beg how can I remove this error, I have been looking for an answer for about a week, you are my hope

**Comment by MichaelFarad** [comment]
> @MichaelFarad what exactly are you trying to do?

I can't fix this error....I'm in the wall, I don't know what to do

**Comment by Lpsd** [comment]
Nothing described on this page can help you immediately - this is a development discussion.

GitHub isn't for general MTA support. If you're experiencing issues please visit our [Discord](https://mtasa.com/discord) and use the #help-support channel.
